packages/components/src/components/flipcard/flipcard.ts

```diff
@@ -0,0 +1,337 @@
+import { css, html } from 'lit';
+import { customElement } from '../../internal/register-custom-element';
+import { property, query } from 'lit/decorators.js';
+import componentStyles from '../../styles/component.styles';
+import cx from 'classix';
+import SolidElement from '../../internal/solid-element';
+
+/**
+ * @summary Flipcard allows for the addition of content/information on both "sides" of the card, through means of a flip animation. Used  to add dynamism and interactivity to a page.
+ * @documentation https://solid.union-investment.com/[storybook-link]/flipcard
+ * @status stable
+ * @since 3.8.0
+ *
+ * @event sd-flip-front - Emmited when the front face of the flipcard is clicked.
+ * @event sd-flip-back - Emmited when the back face of the flipcard is clicked.
+ *
+ * @slot front - The front face of the flipcard.
+ * @slot back - The back face of the flipcard.
+ * @slot mediaFront - An optional media slot which can be as a background. Dependent from gradient variant.
+ * @slot mediaBack - An optional media slot which can be as a background. Dependent from gradient variant.
+ *
+ * @cssparts base - The component's base wrapper.
+ * @cssparts front - The container that wraps the front-side of the flipcard.
+ * @cssparts back - The container that wraps the back-side of the flipcard.
+ * @cssparts frontSlotContainer - The container that wraps the front slot.
+ * @cssparts backSlotContainer - The container that wraps the back slot.
+ * @cssparts mediaFront - The container that wraps the mediaFront slot.
+ * @cssparts mediaBack - The container that wraps the mediaBack slot.
+ * @cssparts frontSecondaryGradient - The container that wraps the secondary gradient of the front side.
+ * @cssparts backSecondaryGradient - The container that wraps the secondary gradient of the back side.
+ *
+ * @cssproperty --name - Description of the flipcard.
+ * @cssproperty --height - Use this property to set the height of the flipcard.
+ */
+
+@customElement('sd-flipcard')
+export default class SdFlipcard extends SolidElement {
+  @query('[part="back"]') back: HTMLElement;
+  @query('[part="front"]') front: HTMLElement;
+
+  /**
+   * Determines the activation type of the flipcard.
+   */
+  @property({ reflect: true }) activation: 'click-only' | 'click-and-hover' = 'click-and-hover';
+
+  /**
+   * Allows the flipcard to flip vertically or horizontally.
+   */
+  @property({ reflect: true, attribute: 'flip-direction' }) flipDirection: 'horizontal' | 'vertical' = 'horizontal';
+
+  /** Determines the variant of the front face of the flipcard. */
+  @property({ type: String, reflect: true })
+  frontVariant:
+    | 'primary'
+    | 'primary-100'
+    | 'gradient-light-top'
+    | 'gradient-light-bottom'
+    | 'gradient-dark-top'
+    | 'gradient-dark-bottom' = 'primary';
+
+  /** Determines the variant of the back face of the flipcard. */
+  @property({ type: String, reflect: true }) backVariant:
```

Comment:
Didn't we agree that we add an empty one as default?

Reply:
See https://github.com/solid-design-system/solid/pull/1121#issuecomment-2146984828